### PR TITLE
Test off curve pubkey

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

No test that pubkeys which aren't on the ed25519 curve cause signature verification to fail.

#### Summary of Changes

Add one

### CODE REVIEW

File: sdk/Cargo.toml [hunk 1]
```diff
@@ -60,6 +60,7 @@ solana-sdk-macro-frozen-abi = { path = "macro-frozen-abi", version = "1.3.0" }
 rustversion = "1.0.3"
 
 [dev-dependencies]
+curve25519-dalek = { version = "2", default-features = false }
```
Comment (jackcmay) [inline sdk/Cargo.toml:63]: Why `default_features = false`?  Cut down on bloat?

File: sdk/src/signature.rs [hunk 2]
```diff
@@ -61,16 +61,18 @@ impl Signature {
         Self(GenericArray::clone_from_slice(&signature_slice))
     }
 
+    pub(self) fn verify_verbose(
```
Comment (jackcmay) [inline sdk/src/signature.rs:64]: You didn't introduce this but I would change the variable names here from using 'pubkey' to 'publickey' to make it clear that we are referring to an `ed25519_dalek::PublicKey` and not Solana's `Pubkey`

File: sdk/Cargo.toml (hunk 1 above)
Comment (t-nelson) [reply sdk/Cargo.toml:63]: I just snagged the dep line out of `ed25519-dalek`s `Cargo.toml`. Happy to drop it if you like

**Comment by t-nelson** [comment]
@jackcmay I went ahead and addressed both

**Comment by sakridge** [comment]
Would be nice if this is at the level where the GPU path is also tested.

**Pull request merged**